rosidl_typesupport_opendds_cpp/rosidl_typesupport_opendds_cpp/__init__.py

```diff
@@ -73,6 +96,42 @@ def generate_dds_opendds_cpp(
             print("'%s' failed to generate the expected files for '%s/%s'" %
                   (idl_pp, pkg_name, msg_name), file=sys.stderr)
 
+        if "/srv/" in idl_file:
+            generated_idl_file = output_path + "/" + msg_name + "_RequestResponse.idl"
+            input_idl_path = os.path.dirname(idl_file)
+
+            try:
+                cmd = [idl_pp, generated_idl_file, "-Lc++11", "-o", output_path, "-Sa", "-St", "-Cw", "--no-dcps-data-type-warnings"]
+                for include_dir in include_dirs:
+                    cmd += ['-I', include_dir]
+                cmd += ['-I', input_idl_path]
+                cmd += ['-I', rpc_file_path]
+                subprocess.check_call(cmd)
+            except subprocess.CalledProcessError as e:
+                raise RuntimeError('OpenDDS compiler failed to generate the expected service files')
+            
+            any_missing = False
+            for suffix in ['_RequestResponseC.h', '_RequestResponseTypeSupportImpl.cpp', '_RequestResponseTypeSupportImpl.h', '_RequestResponseTypeSupport.idl']:
+                filename = os.path.join(output_path, msg_name + suffix)
+                if not os.path.exists(filename):
+                    any_missing = True
+                    break
+            if any_missing:
+                print("'%s' failed to generate the expected files for '%s/%s'" %
+                      (idl_pp, pkg_name, msg_name), file=sys.stderr)
+
+            try:
+                generated_idl_file = output_path + "/" + msg_name + "_RequestResponseTypeSupport.idl"
+                cmd = ["tao_idl", generated_idl_file, "-o", output_path, "-I$TAO_ROOT", "-I$DDS_ROOT", "-I/opt/OpenDDS", "--idl-version",  "4", "-SS", "-Sa", "-St", "-Wb,pre_include=ace/pre.h", "-Wb,post_include=ace/post.h", "--unknown-annotations", "ignore"]
```

Comment:
I ran this branch against the system tests. I think you'll need the `-Cw` flag here.
```suggestion
                cmd = ["tao_idl", generated_idl_file, "-o", output_path, "-I$TAO_ROOT", "-I$DDS_ROOT", "-I/opt/OpenDDS", "--idl-version",  "4", "-SS", "-Sa", "-St", "-Cw", "-Wb,pre_include=ace/pre.h", "-Wb,post_include=ace/post.h", "--unknown-annotations", "ignore"]
```

Reply:
Done